addons/product/models/product.py

```diff
@@ -395,17 +395,36 @@ def _name_get(d):
         self.check_access_rule("read")
 
         result = []
+
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+        # Use `load=False` to not call `name_get` for the `product_tmpl_id`
+        self.sudo().read(['name', 'default_code', 'product_tmpl_id', 'attribute_value_ids'], load=False)
+
+        product_template_ids = [product.product_tmpl_id.id for product in self]
```

Comment:
Comment1: Why an api method is not used here `self.mapped('product_tmpl_id').ids`?
Comment2: What about if the products (variants) are using the same template?